service/history/replication/task_ack_manager.go

```diff
@@ -625,6 +640,24 @@ func (t *taskAckManagerImpl) generateHistoryReplicationTask(
 	)
 }
 
+func (t *taskAckManagerImpl) getBatchSize() int {
+
+	t.taskLock.Lock()
+	taskLatency := int64(time.Now().Sub(t.lastTaskCreationTime) / time.Second)
+	t.taskLock.Unlock()
```

Comment:
Trying to understand how useful this lock is: it feels like having and not having the lock here are the same to me. Could you you explain a bit what it is achieving here?

Are the time assignments in go not atomic that if we try to read its value while assigning somewhere else we get a crash?

Reply:
Most likely it will not be used. I added it to prevent concurrent calls to the method. Updated it with atomic.Value